### PR TITLE
testament: fix #16829, fix partially #16830

### LINKED ISSUE #16829: testament regression
When using testament on external project
https://github.com/planety/prologue/runs/1766905861

progress[all]: 14/15 starting: cat: megatest
Error: unhandled exception: categories.nim(710, 16) cat.string in whiteListedDirs Invalid category specified: 'ic' not in whilelist: ["deps", "htmldocs", "pkgs"] [AssertionDefect]
'/home/runner/.choosenim/toolchains/nim-#devel/bin/testament' --nim:/home/runner/.nimble/bin/nim pcat megatest
joinable specs: 36

Ref #16162

See https://github.com/planety/prologue

### PR DESCRIPTION
**Pull request by timotheecour**
* fix #16829
* fix partially #16830 (most remaining issues can be fixed similarly via `isNimRepoTests`)
* closes #16827 (or, that PR can be changed to only do `pkg2 "prologue", "nimble tests"`)

### CODE REVIEW

File: testament/testament.nim [hunk 1]
```diff
@@ -86,6 +93,10 @@ var targetsSet = false
 
 proc isSuccess(input: string): bool =
   # not clear how to do the equivalent of pkg/regex's: re"FOO(.*?)BAR" in pegs
+  # note: this doesn't handle colors, eg: `\e[1m\e[0m\e[32mHint:`; while we
+  # could handle colors, there would be other issues such as handling other flags
+  # that may appear in user config (eg: `--listFullPaths`).
+  # Passing`XDG_CONFIG_HOME= ` is one way to avoid interefence from user config.
```
Comment (ringabout) [inline testament/testament.nim]:
```suggestion
  # Passing `XDG_CONFIG_HOME= ` is one way to avoid interefence from user config.
```

Comment (ringabout) [reply testament/testament.nim]: Please give notes or link for `XDG_CONFIG_HOME= `

**Review by ringabout — APPROVED** [review]
thanks!

File: testament/testament.nim (hunk 1 above)
Comment (timotheecour) [reply testament/testament.nim]: fixed

**Pull request merged**